src/doc/book/syntax-index.md

```diff
@@ -161,6 +161,7 @@
 
 <!-- Various things involving parens and tuples -->
 
+* `!`: empty never type.
```

Comment:
This wording could be confused for "a type that is never empty." If we're using both terms, I'd go with something like

> empty (or "never") type